engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -104,10 +104,8 @@ protected void writeObject(TailCall.TailPosition obj, Output out) throws IOExcep
     @Override
     protected TailCall.TailPosition readObject(Input in)
         throws IOException, ClassNotFoundException {
-      var b = in.readBoolean();
-      return b
-          ? org.enso.compiler.pass.analyse.TailCall$TailPosition$Tail$.MODULE$
-          : org.enso.compiler.pass.analyse.TailCall$TailPosition$NotTail$.MODULE$;
+      in.readBoolean();
```

Comment:
Reading a boolean to keep compatibility with previous format, but throwing it away and always returning `Tail`? I think we should increase the ID of `PersistTail` and treat it is singleton - e.g. don't store the boolean at all. Probably simple
```java
@Persistable(clazz=TailCall.Tail.class, id=1102+x)
```
should generate the right code for the singleton persistance.

Reply:
Done in 862f8ee65e2375568c38ffad7b375117deb465e7